overlord/snapstate/component.go

```diff
@@ -80,6 +80,7 @@ func InstallComponents(ctx context.Context, st *state.State, names []string, inf
 		kmodSetup = st.NewTask("prepare-kernel-modules-components", fmt.Sprintf(
 			i18n.G("Prepare kernel-modules components for %q%s"), info.InstanceName(), info.Revision,
 		))
+		kmodSetup.Set("snap-setup-task", setupSecurity.ID())
```

Comment:
shouldn't this use snapSetupTaskID ?

Reply:
`snapSetupTaskID` isn't in scope here, that is a param to `doInstallComponent`. In this case though, when we call `doInstallComponent` we use `setupSecurity.ID()` as that parameter.